rotkehlchen/accounting/constants.py

```diff
@@ -1,206 +1,264 @@
         label='receive',
         icon='mdi-arrow-down',
         color='green',
+        direction=EventDirection.IN,
     ),
     EventCategory.SWAP_OUT: EventCategoryDetails(
         label='swap out',
         icon='mdi-arrow-u-right-bottom',
+        direction=EventDirection.OUT,
     ),
     EventCategory.SWAP_IN: EventCategoryDetails(
         label='swap in',
         icon='mdi-arrow-u-left-top',
         color='green',
+        direction=EventDirection.IN,
     ),
     EventCategory.APPROVAL: EventCategoryDetails(
         label='approval',
         icon='mdi-lock-open-outline',
+        direction=EventDirection.NEUTRAL,
     ),
     EventCategory.DEPOSIT: EventCategoryDetails(
         label='deposit',
         icon='mdi-arrow-expand-up',
         color='green',
+        direction=EventDirection.OUT,
     ),
     EventCategory.WITHDRAW: EventCategoryDetails(
         label='withdraw',
         icon='mdi-arrow-expand-down',
+        direction=EventDirection.IN,
     ),
     EventCategory.AIRDROP: EventCategoryDetails(
         label='airdrop',
         icon='mdi-airballoon-outline',
+        direction=EventDirection.IN,
     ),
     EventCategory.BORROW: EventCategoryDetails(
         label='borrow',
         icon='mdi-hand-coin-outline',
+        direction=EventDirection.IN,
     ),
     EventCategory.REPAY: EventCategoryDetails(
         label='repay',
         icon='mdi-history',
+        direction=EventDirection.OUT,
     ),
     EventCategory.DEPLOY: EventCategoryDetails(
         label='deploy',
         icon='mdi-swap-horizontal',
+        direction=EventDirection.NEUTRAL,
     ),
     EventCategory.BRIDGE_IN: EventCategoryDetails(
         label='bridge',
         icon='mdi-arrow-expand-up',
         color='red',
+        direction=EventDirection.OUT,
     ),
     EventCategory.BRIDGE_OUT: EventCategoryDetails(
         label='bridge',
         icon='mdi-arrow-expand-down',
         color='green',
+        direction=EventDirection.IN,
     ),
     EventCategory.GOVERNANCE: EventCategoryDetails(
         label='governance',
         icon='mdi-bank',
+        direction=EventDirection.NEUTRAL,
     ),
     EventCategory.DONATE: EventCategoryDetails(
         label='donate',
         icon='mdi-hand-heart-outline',
+        direction=EventDirection.OUT,
     ),
     EventCategory.RECEIVE_DONATION: EventCategoryDetails(
         label='receive donation',
         icon='mdi-hand-heart-outline',
+        direction=EventDirection.IN,
     ),
     EventCategory.RENEW: EventCategoryDetails(
         label='renew',
         icon='mdi-calendar-refresh',
+        direction=EventDirection.OUT,
     ),
     EventCategory.PLACE_ORDER: EventCategoryDetails(
         label='place order',
         icon='mdi-briefcase-arrow-up-down',
+        direction=EventDirection.NEUTRAL,
     ),
     EventCategory.TRANSFER: EventCategoryDetails(
         label='transfer',
         icon='mdi-swap-horizontal',
+        direction=EventDirection.IN,
```

Comment:
So here there is a problem. `EventCategory.TRANSFER` is not always an `IN` event. The only place that maps this is `EventType.TRANSFER`/EventSubType.NONE`. Only place this happens explicitly is here: https://github.com/rotki/rotki/blob/e047af1772f470d71217daee97f5c50cb4158d64/rotkehlchen/chain/ethereum/modules/uniswap/v2/common.py#L172-L177

And yeah there it's a refund where we get something back. So the author there thought to use `TRANSFER/NONE`as that is the combo that means transfering between own accounts, so not taxable.

Now the problem though is that this is not always IN. It can also be OUT. It depends on how you see it. The basic decoders also create ETH value or token transfers with `TRANSFER/NONE` if both from and to are tracked addresses.

So the problem here is in those cases what is the direction?

Reply:
Since transfers between owned accounts only get decoded once what I've done is:

- Create the neutral category INTERNAL_MOVEMENT that is meant to be used by the user for movements between their accounts. Is not limited to evm chains but can happen between exchanges and others
- The transfers between tracked accounts are now INTERNAL_MOVEMENT
- The combination TRANSFER/NONE is still and IN